src/azure-cli-core/azure/cli/core/auth/adal_authentication.py

```diff
@@ -27,18 +28,29 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
             # If available, use resource provided by SDK
             self.resource = resource
         self.set_token()
-        # Managed Identity token entry sample:
+        # VM managed identity endpoint 2018-02-01 token entry sample:
+        # curl "http://169.254.169.254:80/metadata/identity/oauth2/token?resource=https://management.core.windows.net/&api-version=2018-02-01" -H "Metadata: true"
         # {
         #     "access_token": "eyJ0eXAiOiJKV...",
         #     "client_id": "da95e381-d7ab-4fdc-8047-2457909c723b",
         #     "expires_in": "86386",
         #     "expires_on": "1605238724",
         #     "ext_expires_in": "86399",
         #     "not_before": "1605152024",
-        #     "resource": "https://management.azure.com/",
+        #     "resource": "https://management.core.windows.net/",
         #     "token_type": "Bearer"
         # }
-        return AccessToken(self.token['access_token'], int(self.token['expires_on']))
+
+        # App Service managed identity endpoint 2017-09-01 token entry sample:
+        # curl "${MSI_ENDPOINT}?resource=https://management.core.windows.net/&api-version=2017-09-01" -H "secret: ${MSI_SECRET}"
+        # {
+        #     "access_token": "eyJ0eXAiOiJKV...",
+        #     "expires_on":"11/05/2021 15:18:31 +00:00",
```

Comment:
This example response was retrieved on a Linux App Service.

It has been reported that managed identity on Windows App Service returns `expires_on` in a different format:

```
"expires_on":"8/5/2023 9:13:43 AM +00:00"
```

causing Azure CLI to fail with 

```
time data '8/3/2023 9:26:33 AM +0000' does not match format '%m/%d/%Y %H:%M:%S %z'
```